### ISSUE TITLE
Several warnings building on .Net Core

### ISSUE THREAD

**Issue by galvesribeiro** [Azure/Azure-Functions#559]
```
/Users/guto/dev/repos/GiroPlatform/test/InfrastructureTests/InfrastructureTests.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/src/GP.Calculator/GP.Calculator.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/src/GP.Accounting/GP.Accounting.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/src/GP.Infrastructure/GP.Infrastructure.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/src/GP.Stone/GP.Stone.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1'instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/test/CalculatorTests/CalculatorTests.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/test/AccountingTests/AccountingTests.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/test/StoneTests/StoneTests.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/src/GP.Merchants/GP.Merchants.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/dev/repos/GiroPlatform/test/MerchantsTests/MerchantsTests.csproj : warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
/Users/guto/.nuget/packages/microsoft.net.sdk.functions/1.0.6/build/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets(32,5): warning :  [/Users/guto/dev/repos/GiroPlatform/src/GP.Infrastructure/GP.Infrastructure.csproj]
/Users/guto/.nuget/packages/microsoft.net.sdk.functions/1.0.6/build/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets(32,5): warning :  [/Users/guto/dev/repos/GiroPlatform/src/GP.Stone/GP.Stone.csproj]
/Users/guto/.nuget/packages/microsoft.net.sdk.functions/1.0.6/build/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets(32,5): warning :  [/Users/guto/dev/repos/GiroPlatform/src/GP.Accounting/GP.Accounting.csproj]
/Users/guto/.nuget/packages/microsoft.net.sdk.functions/1.0.6/build/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets(32,5): warning :  [/Users/guto/dev/repos/GiroPlatform/src/GP.Calculator/GP.Calculator.csproj]
/Users/guto/.nuget/packages/microsoft.net.sdk.functions/1.0.6/build/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets(32,5): warning :  [/Users/guto/dev/repos/GiroPlatform/src/GP.Merchants/GP.Merchants.csproj]
    15 Warning(s)
```

Those warnings came out after update to latest SDK 1.0.6. When I was on 1.04, there were multiple errors as stated [here](https://github.com/Azure/Azure-Functions/issues/98#issuecomment-338274213).

I wonder if we can remove that somehow.

Thanks!

**Comment by ahmelsayed** [comment]
@fabiocav  I believe those warning are expected because we're using some kind of shim to still support some types from ASP.NET? @fabiocav  would definitely know more.

**Comment by MV10** [comment]
Still happening as of 1.0.7 on Microsoft.AspNet.WebApi.Client.

![fail](https://i.imgur.com/tIkMjk4.jpg)

**Comment by jeffputz** [comment]
I just encountered this today, trying to use .NET Core libraries with functions. Same error on the package restore:
`warning NU1701: Package 'Microsoft.AspNet.WebApi.Client 5.2.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.`

**Comment by fabiocav** [comment]
This is a harmless warning and should have no negative impact. For the time being, please feel free to suppress.

**Comment by jeffputz** [comment]
I still can't build the project, even following the documentation to the letter. I can't make a Core function build or run locally.

**Comment by fabiocav** [comment]
What build errors are you seeing? Can  you share the project on a repo here?

**Comment by tparvi** [comment]
1. Create new "Azure Functionv v2 (.NET Core)" project in Visual Studio with Time trigger
2. Create new .NET Core 2 class library
3. Reference newly created .NET Core 2 class library from the function project

Result is error during build:
`Project '..\ClassLibrary1\ClassLibrary1.csproj' targets 'netcoreapp2.0'. It cannot be referenced by a project that targets '.NETStandard,Version=v2.0'.	FunctionApp1
`

I have updated to 1.0.7. Should I be able to reference .NET Core 2 class libraries? I understood that the ".NET Core" support for Azure functions means this.

**Comment by fabiocav** [comment]
@tparvi, the Azure Functions project targets .NET Standard 2.0, so you'll be able to consume .NET Standard class libraries (which can also be consumed by your .NET Core projects). When creating your library, please select .NET Standard as opposed to .NET Core.

**Comment by benc-uk** [comment]
I see this warning too, using the created template HttpTrigger C# function. No other changes, no other libraries.
I'm using VS Code, and the Functions extension, but that is unrelated to the issue

**Comment by Mike-EEE** [comment]
FWIW, I get two warnings in my case, one being blank with no details/text provided:

![](https://content.screencast.com/users/Mike-EEE/folders/Jing/media/fca0520e-6321-470d-a41d-78516954a2b3/2018-03-19_0526.png)

**Comment by teamboyd** [comment]
@Mike-EEE your second warning is due to your FunctionApp1 actually not existing in Azure yet.  It's not due to a dependency warning like the first one.  Does this make sense?

**Comment by Mike-EEE** [comment]
Ok cool @teamboyd, thank you for that.  Yes, that makes sense.  What doesn't make sense is that I have to report it to Visual Studio feedback who then points me over here to GitHub to try to track it down. 😄 

Still, even in your explanation, it doesn't sound like you know _exactly_ what it is.  Do you happen to see this when creating a new Azure Function app from scratch?  That is all I did and was met with these two warnings.

I understand that it's not due to a dependency, but now I wonder if this should be tracked in another issue altogether?  All the warnings here are due to dependencies, if I understand correctly.  

Or perhaps am I the only one experiencing this for some odd reason?  In any case, it is very odd that no message is given, which only adds to the issue.

**Comment by teamboyd** [comment]
Sorry about the scribbles but here is my 2nd warning with specifics.  I would assume your warning is similar but you may not have any values entered into your FunctionApp1 attributes and this is possibly why your description is blank.  Either way, a blank description is a Microsoft bug in my opinion.  :)

![image](https://user-images.githubusercontent.com/636811/37982344-3314f3ea-31a5-11e8-8c2d-ae3da1427fea.png)

**Comment by Mike-EEE** [comment]
Ah!  OK... perhaps this is due to the type of Azure Function I am creating?  It seems you are creating a queue trigger.  Can you try creating a http trigger project and seeing if you get the blank message?

**Comment by fabiocav** [comment]
This warning has been addressed on the latest package updates done yesterday (beta5 packages). An updated SDK package will be released by the end of the week and once that is done, those pesky warnings will be gone.

**Comment by fabiocav** [comment]
I don't see the empty warning when using the updated packages but will validate once we have the final SDK packaged up.

**Comment by teamboyd** [comment]
It will fix both warnings?  The empty warning and the other pesky version incompatibility warning?

**Comment by fabiocav** [comment]
Answered above. Will validate that this results in a clean build.

**Comment by Mike-EEE** [comment]
As they say @fabiocav the reward for hard work is more hard work. 😆 

By chance, are you aware of any additional installs that are required starting from `1.0.11`?  The reason I ask is that now while the warnings are gone, when I take the next step and F5 my Azure Function, it results in an error as described in https://github.com/Azure/azure-functions-host/issues/2605.  Since I seem to be the only one that is encountering this (very obvious and disruptive) problem, it's making me think that I am missing an install of some sort.  

I have reverted back it `1.0.10` in the meantime, but the warnings are back. 😞 

I've been searching around and nothing recent strikes me, so I am thinking it is localized/install related. Any guidance/assistance/insight you (or anyone) can lend would be greatly appreciated.  I am a total newb here and have only been exploring this for about a week or so now.  Thanks in advance!

**Comment by Mike-EEE** [comment]
Nevermind @fabiocav  and all... I figured it out (for now):
https://github.com/Azure/azure-functions-host/issues/2605#issuecomment-377583107

I am feeling all Azure Function Host certified and all. 😛   Onto the next problem. 😛

**Comment by rhummelmose** [comment]
I am getting these warnings in Functions v2, newest runtime. Can you verify this?
